lnwire/lnwire.go

```diff
@@ -326,6 +335,58 @@ func writeElement(w io.Writer, element interface{}) error {
 			return err
 		}
 
+	case *torsvc.OnionAddress:
+		if len(e.HiddenService) == 22 {
+			// v2 hidden service
+			var descriptor [1]byte
+			descriptor[0] = uint8(v2OnionAddr)
+			if _, err := w.Write(descriptor[:]); err != nil {
+				return err
+			}
+
+			// unbase32 the v2 hidden service string, truncating
+			// the .onion suffix
+			data, err := encoding.DecodeString(e.String()[:16])
+			if err != nil {
+				return err
+			}
+			if _, err := w.Write(data); err != nil {
+				return err
+			}
+
+			// write the port
+			var port [2]byte
+			binary.BigEndian.PutUint16(port[:], uint16(e.Port))
+			if _, err := w.Write(port[:]); err != nil {
+				return err
+			}
+
+		} else {
+			// v3 hidden service
+			var descriptor [1]byte
+			descriptor[0] = uint8(v3OnionAddr)
+			if _, err := w.Write(descriptor[:]); err != nil {
+				return err
+			}
+
+			// unbase32 the v3 hidden service string, truncating
+			// the .onion suffix
+			data, err := encoding.DecodeString(e.String()[:56])
```

Comment:
these magic numbers should probably be extracted into exported constants in `torsvc`